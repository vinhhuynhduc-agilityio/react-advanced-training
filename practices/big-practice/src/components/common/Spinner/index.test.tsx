import { render } from '@testing-library/react';
import { Spinner } from '.';

describe('Spinner Component', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Spinner />);
    expect(container).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    const { container } = render(<Spinner />);
    expect(container.firstChild).toBeTruthy();
  });

  it('should render the spinner correctly', () => {
    const { container } = render(<Spinner />);

    // Check if the spinner div is rendered
    const spinnerDiv = container.querySelector('div');
    expect(spinnerDiv).toBeInTheDocument();

    // Check if the spinner div has the correct classes
    expect(spinnerDiv).toHaveClass(
      'flex items-center justify-center w-full h-full'
    );

    // Check for the inner spinning element
    const spinnerInner = container.querySelector(
      '.w-12.h-12.border-4.border-gray-300.border-t-blue-500.rounded-full.animate-spin'
    );
    expect(spinnerInner).toBeInTheDocument();
  });
});
