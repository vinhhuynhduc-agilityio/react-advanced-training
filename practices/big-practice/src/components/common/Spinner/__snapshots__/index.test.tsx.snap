// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component should match the snapshot 1`] = `
<div>
  <div
    class="flex items-center justify-center w-full h-full"
  >
    <div
      class="w-12 h-12 border-4 border-gray-300 border-t-blue-500 rounded-full animate-spin"
    />
  </div>
</div>
`;
